### DISCUSSION TITLE
Welcome to create-nes-game Discussions!

### DISCUSSION THREAD

**Discussion by potatolain** [igwgames/create-nes-game#1] [Announcements]
## 👋 Welcome!

I've decided to try out discussions with this repository, to give a way to talk about the project without having to open an issue. You can always reach me in other ways, such as [twitter](https://twitter.com/cppchriscpp) or discord or whatever; this is just another option.

Feel free to:
  * Ask questions you’re wondering about. Ask the whys and the hows if you're curious!
  * Share ideas
  * Engage with other community members.
  * Welcome others and be open-minded. (In fact, please do this!)

I'm not writing up any strict rules yet, but know that this is not a place for insults, racism, homophobia, sexism, transphobia, etc etc. 

I hope to hear from you, and I hope this thing helps lower the barrier of entry to making games even more!

**Comment by RiffRunner** [comment]
how can I include nesdoug's library (nesdoug.h)?

**Reply by potatolain** [reply]
Hey there! 

So I don't have a built-in way to do it right now, though that's probably a worthwhile thing for me to look at adding. You'll have to add it to generated create-nes-game code if you want it. That's not super hard, though! I'm going off his hello world example, here: https://github.com/nesdoug/01_Hello - though any later ones should work the same way.

1. Create a new game if you haven't yet. Make sure to pick C as your language, use `famitone2` for your sound library, and _don't include neslib_. (Doug has his own version that's modified, and while it'd probably work with the built-in one, I can't promise it will!)
2. Copy `LIB/nesdoug.h` and `LIB/neslib.h` from the project into `YOUR_GAME/source/c` (you can use a `lib` folder if you'd like, but it's not required)
3. Copy `LIB/nesdoug.s` and `LIB/neslib.s` from the project into `YOUR_GAME/source/assembly` (ditto on `lib` folder)
4. Edit `YOUR_GAME/source/assembly/system-runtime.asm` and add the following two lines: 
    > .include "./neslib.s" ; Adapt this to whatever path you used in step 3, relative to where `system-runtime.asm ` lives
    > .include "./nesdoug.s" 
5. In your C, `#include "nesdoug.h"` and `#include "neslib.h"` and you should more-or-less be ready to go

You may end up also having to replace the generated chr files with ones from the project, and other things like that. Looking at git should give you some hints on what needs to move where.

Let me know if you run into any odd issues
